egg/zoo/emcom_as_ssl/archs.py

```diff
@@ -13,17 +13,21 @@
 from egg.core.gs_wrappers import gumbel_softmax_sample
 
 
-def get_resnet(name: str = "resnet50", pretrained: bool = False):
-    """Loads ResNet encoder from torchvision along with features number"""
-    resnets = {
+def get_pretrained(name: str = "resnet50", pretrained: bool = True):
+    print("Pretrained?", pretrained)
+    print("Model", name)
+    modules = {
         "resnet50": torchvision.models.resnet50(pretrained=pretrained),
         "resnet101": torchvision.models.resnet101(pretrained=pretrained),
         "resnet152": torchvision.models.resnet152(pretrained=pretrained),
+        "vgg16": torchvision.models.vgg16(pretrained=pretrained),
+        "densenet": torchvision.models.densenet161(pretrained=pretrained),
+        "inception": torchvision.models.inception_v3(pretrained=pretrained)
     }
-    if name not in resnets:
+    if name not in modules:
         raise KeyError(f"{name} is not a valid ResNet architecture")
 
-    model = resnets[name]
+    model = modules[name]
     n_features = model.fc.in_features
```

Comment:
This wouldn't work with vgg. 

https://pytorch.org/tutorials/beginner/finetuning_torchvision_models_tutorial.html